### ISSUE TITLE
Formalize FediBlock activity

### ISSUE THREAD

**Issue by jfinkhaeuser** [swicg/activitypub-trust-and-safety#19]
This issue is a final comment on my minor distraction in #14, to move the discussion here. It likely has a lot of overlap with @ThisIsMissEm's FIRES proposal.

The FediBlock hashtag has emerged as the de-facto standard for sharing moderation recommendations in-band (out-of-band recommendations between admins appear to the major other mechanism). It is immediately useful, but suffers from a number of downsides:

- There is no standard nor best practice for what a post with the hashtag should contain.
- Following up on a recommendation is tedious in particular because even if sufficient information is contained to make a judgement, verifying the information's veracity is always necessary.
- As a result of both, it's open to abuse.

The major benefit of the approach is that it is decidedly bottom-up: every community can define their own standards that befit them best, and communities are as overlapping as the people sharing the posts - it leads to very organic, non-prescriptive moderation practices.

The counter-argument to this is that an organization with dedicated resources to ensuring safety can better make trustworthy moderation recommendations than any overwhelmed individual admin ever could.

This proposal aims to satisfy **both** scenarios.

---

I'd like to proposal a formal FediBlock (or ModerationRecommendation) activity, which will come in two scopes, two flavours, and in two communications "channels".

**Scopes**

- Recommendations can be on individual accounts, or...
- ... on entire instances.

The moderation actions available will differ per scope.

**Flavours**

- A free-form flavour should mimic FediBlock largely as-is: it needs to provide a scope, as well as a textual description of the reason. It should furthermore include references to and/or copies of offending behaviour that leads to the recommendation. This includes usage of #15 to explicitly flag the reasons. Additionally, #18 can be used to describe the recommended moderation action; this, however, is necessarily the action that would be taken on the originating server.
- A structured flavour foregoes the description, and most of the above data. Instead, it needs to provide a scope, as well as a list of one or more Flags (which already contain receipts and reasons). The only remaining field that can be added is the recommended moderation action.

**Communications Channels**

- User-to-user, as current FediBlock does. This should be available to non-admin users, and may cover both scopes (however, an instance scope is only actionable by administrators, so an argument can be made to limit this to the account scope). Similarly, both flavours are feasible, but the structured flavour is mostly going to be able to cover Flags raised by the same individual who issues the recommendation.
- Instance-to-instance (effectively: administrator(s) to administrator(s)). The idea would be to strongly prefer (require?) the structured flavour in either scope as administrators have access to the relevant Flags.

---

A few notes:

1. The instance-to-instance, structured flavour effectively implements shared blocklists. I would phrase a FEP such that these activities MUST NOT be automatically be translated into moderation without human review. Blocklists, like FediBlock, can be abused. Automating moderation based on blocklists is an obvious optimization, but it effectively hands over control over one's social bubble to a third party.
2. An organization with dedicated resources for providing safety to the fediverse at large would effectively produce structured, instance-to-instance (i.e. subscribeable) recommendation lists. Since they might be predominantly acting on Flags or information not originating from their own server, an appropriate sharing mechanism is still required -- but the recommendation lists of participating servers would provide this. Such an organization would essentially correlate and review recommendations, and publish their own based on this process.
3. Because of the above, and because of a general notion of consent, I would advise that there be a standard way to make administrators aware of other instance's recommendation lists, without pushing these activities automatically to instances one federates with. This way, consumption of recommendations is opt-in. Note that this may not apply the same to free-form, user-to-user activities, but also here an opt-in/opt-out mechanism should be provided.
4. User interfaces to produce recommendations from (a collection of) flags should be provided.
5. Implementing a recommendation should optionally/always (?) forward the recommendation to the own published list. This may require a link to the originating recommendation, as the implementation will likely overwrite the moderation action based on locally available choices.
6. Since Flags contain (supposedly?) anonymised actor information, a recommendation with the actor information in plain text effectively de-anonymises them. Instance moderation recommendations do not do this.

**Comment by ThisIsMissEm** [comment]
I need to read this proposal more fully, but I would be inclined to say that we **should not** formalise the existing hashtag & it's practices in anyway, since it is often misused, abused, and lacks proper moderation — i.e., regular people using the fediverse regularly take moderation issues to that hashtag rather than letting their moderators deal with it.

That said, I think there could be interest in defining a mechanism to Flag an instance & share it with given servers, however this requires servers/instances to be formally defined, something which is not currently done in the actor-centric ActivityPub Specification

**Comment by ThisIsMissEm** [comment]
I will also point out the "scope of work" section in the readme: https://github.com/swicg/activitypub-trust-and-safety?tab=readme-ov-file#scope-of-work

**Comment by bumblefudge** [comment]
> It should come after defining moderation actions

I agree that what you're describing is work worth doing, and best done after a report on FIRES has shipped; the questions this begs is who is doing it? If there are people volunteering and chomping at the bit to formalize and upgrade today's block Activity, maybe start with a FEP and see if the work can be sketched out in parallel to see if it might be a separate work item of the TF?

**Comment by bumblefudge** [comment]
note that the CG is trying to upgrade and formalize a little it's own workflow; see for instance this [DRAFT workflow](https://github.com/swicg/potential-charters/blob/main/stage-process.md) to give you an idea of how work items can be "iteratively formalized" based on how political/loadbearing they are, and how long it takes to find colleagues willing to really hash out/codesign with you 😀

**Comment by bumblefudge** [comment]
Oh, and another thing that just occurred to me is that FediBlock lives in an off-standard grey area, which doesn't really matter for assessing the _urgency_ or _utility to today's users_ of upgrading it (or replacing it) as a mechanism, but _does_ impact how easily the "standardization people" (a lawyerly and slow-moving species) can design solutions around it.  Namely, FediBlocks aren't _user_ blocks but "_instance_" blocks, which take as their target "Instance Actors", which are a Mastodonism not present in the W3C spec ActivityPub at all.  So we can't quite ActivityPub-behavior our way out to an official recommendation since Instances aren't a valid Type defined in ActivityPub at all! The long-deferred, slightly political topic of how to model (and protocolize behavior around) new Actor types (Relays are also a novel Actor type that urgently needs to be formalized a little, as are _external_ Moderation services like FIRES) is upstream of FediBlocks that rely on those types, so this would also be something of a precondition to a CG report telling people how to implement moderation/T&S behaviors specific to those types.

**Comment by ThisIsMissEm** [comment]
> FediBlocks aren't user blocks but "instance" blocks, which take as their target "Instance Actors"

They're not even targeting the "instance actor" but the hostname the instance is on, potentially not even the full hostname either, but a suffix of it.

**Comment by ThisIsMissEm** [comment]
@dariusk any thoughts on closing this? I don't think formalising a "ModerationActivity" or the way FediBlock is used today would be productive, but instead there is likely a side-channel through which moderators may choose to share such information, whether that's FIRES or CSV files or something else.

I don't think this is something that this taskforce should define in the near to medium term future, perhaps in the very long term future.

**Comment by dariusk** [comment]
I think that formalizing FediBlock is not something that should come from the task force. FediBlock has always been an organic community thing and has been characterized by its ability to be fluid in what it is and what it means. Formalization would only ossify FediBlock as "whatever it is as of (date)". I'm in favor of closing this issue.

**Comment by ThisIsMissEm** [comment]
(sorry, accidentally closed this as completed instead of not planned.
